### PR TITLE
john's query test

### PR DESCRIPTION
**Pull request by javasoze**

### CODE REVIEW

File: load_nginx_json.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import datetime
+import json
+import time
+from datetime import timezone
+from datetime import datetime, timedelta
+from locust.contrib.fasthttp import FastHttpLocust
+from nginx_query_seed import generate_reltime
+from nginx_query_seed import generate_query
+
+QUERY_TEMPLATE = """{
```
Comment (keitaf) [inline load_nginx_json.py:11]: remove if not used?

File: load_nginx_json.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    "useApproximation": false,
+    "fetchSchema": false,
+    "disableHighlight": true,
+    "continuationToken": null,
+    "rankingMode": 0,
+    "debugMode": 0
+}"""
+
+def get_query(querystr: str, table: str, starttime: int, endtime: int, sort: bool = False) -> str:
+    with open('request.json') as f:
```
Comment (keitaf) [inline load_nginx_json.py:40]: it's wasteful to load json file every time.

File: load_nginx_json.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                                                         endtime=endtime),
+                           'high_freq_sort': get_query(querystr='333', table='logs', starttime=starttime,
+                                                       endtime=endtime, sort=True),
+                           'low_freq_nosort': get_query(querystr='FireFox', table='logs', starttime=starttime,
+                                                        endtime=endtime),
+                           'low_freq_sort': get_query(querystr='Safari', table='logs', starttime=starttime,
+                                                      endtime=endtime, sort=True)
+                           }
+
+time_last_in_min=1200
```
Comment (keitaf) [inline load_nginx_json.py:72]: what's this?

File: nginx_query_seed.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+statuses=["304", "204", "202", "302"]
+
+def generate_query():
+    keyword = keywords[random.randint(0, len(keywords)-1)]
+    host = hosts[random.randint(0, len(hosts) - 1)]
+    status = statuses[random.randint(0, len(statuses) - 1)]
+    return '({0}) AND (host:"{1}")'.format(keyword, host)
+
+def generate_reltime(range_min):
```
Comment (keitaf) [inline nginx_query_seed.py:19]: range_min isn't used?

File: request.json [hunk 5]
```diff
@@ -20,10 +20,8 @@
     "aggregations": {},
     "useApproximation": false,
     "ctx": "00293a3e-111d-4b69-b303-c03b678cec87",
-    "fetchSchema": true,
-    "timeoutMillis": 1000,
```
Comment (keitaf) [inline request.json:24]: I guess you better set this timeout.

File: nginx_query_seed.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    endtime = int(time.time() * 1000)
+    starttime = int((datetime.utcnow() - timedelta(minutes=15)).replace(tzinfo=timezone.utc).timestamp()) * 1000
```
Comment (sergeyOkuntsev) [inline nginx_query_seed.py:21]: looks like expected to be used as a parameter for timedelta

File: nginx_query_seed.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+hosts=["94.222.333.118", "972.222.333.28", "529.222.333.330", "124.222.333.953", "40.222.333.50", "78.222.333.651"]
+
+statuses=["304", "204", "202", "302"]
+
+def generate_query():
+    keyword = keywords[random.randint(0, len(keywords)-1)]
+    host = hosts[random.randint(0, len(hosts) - 1)]
+    status = statuses[random.randint(0, len(statuses) - 1)]
+    return '({0}) AND (host:"{1}")'.format(keyword, host)
```
Comment (sergeyOkuntsev) [inline nginx_query_seed.py:17]: starting from version 3.6  f strings could be used.  f'({keyword}) AND (host:"{host}")'

File: nginx_query_seed.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+def generate_reltime(range_min):
+    endtime = int(time.time() * 1000)
+    starttime = int((datetime.utcnow() - timedelta(minutes=15)).replace(tzinfo=timezone.utc).timestamp()) * 1000
+    return starttime, endtime
+
+print (generate_query())
+start,end = generate_reltime(15)
+
+print ("start: {0}, end: {1}".format(str(start),str(end)))
```
Comment (sergeyOkuntsev) [inline nginx_query_seed.py:27]: could be print(f"start: {start}, end: {end}")

File: load_nginx_json.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    @locust.seq_task(3)
+    def high_freq2(self):
+        start, end = generate_reltime(time_last_in_min)
+        req = get_query(querystr="222", table='logs', starttime=start, endtime=end)
+        # print (req)
+        response = self.client.post("/query", data=req, headers=self.headers)
+        if response.status_code == 200:
+            res = json.loads(response.content)
+            # print (res)
+            print("numhits (222): {}, took: {}".format(res['numHits'], res['latencyInMillis']))
```
Comment (sergeyOkuntsev) [inline load_nginx_json.py:110]: print(f"numhits (222): {res['numHits']}, took: {res['latencyInMilis']}")

File: load_nginx_json.py (hunk 3 above)
Comment (javasoze) [reply load_nginx_json.py]: default time range, lazy config for the test

File: nginx_query_seed.py (hunk 4 above)
Comment (javasoze) [reply nginx_query_seed.py]: fixed

File: request.json (hunk 5 above)
Comment (javasoze) [reply request.json:24]: why? I thought we want to measure the max latency?

File: nginx_query_seed.py (hunk 6 above)
Comment (javasoze) [reply nginx_query_seed.py]: fixed

File: nginx_query_seed.py (hunk 7 above)
Comment (javasoze) [reply nginx_query_seed.py:17]: nice!

**Pull request merged**